Build on OS X is broken

```
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/GasMeter.cpp:169:22: error: no member named 'getNextNode' in 'llvm::Argument'
        auto cost = gasPtr->getNextNode();
                    ~~~~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Ext.cpp:67:13: error: no matching member function for call to 'SetInsertPoint'
                m_builder.SetInsertPoint(allocaIt);
                ~~~~~~~~~~^~~~~~~~~~~~~~
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:91:8: note: candidate function not viable: no known conversion from 'llvm::ilist_iterator<llvm::Instruction>' to 'llvm::BasicBlock *' for 1st argument
  void SetInsertPoint(BasicBlock *TheBB) {
       ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:98:8: note: candidate function not viable: no known conversion from 'llvm::ilist_iterator<llvm::Instruction>' to 'llvm::Instruction *' for 1st argument
  void SetInsertPoint(Instruction *I) {
       ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:107:8: note: candidate function not viable: requires 2 arguments, but 1 was provided
  void SetInsertPoint(BasicBlock *TheBB, BasicBlock::iterator IP) {
       ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Array.cpp:37:25: error: no member named 'getNextNode' in 'llvm::Argument'
        auto value = arrayPtr->getNextNode();
                     ~~~~~~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Array.cpp:87:25: error: no member named 'getNextNode' in 'llvm::Argument'
        auto index = arrayPtr->getNextNode();
                     ~~~~~~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Array.cpp:111:25: error: no member named 'getNextNode' in 'llvm::Argument'
        auto index = arrayPtr->getNextNode();
                     ~~~~~~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Array.cpp:133:25: error: no member named 'getNextNode' in 'llvm::Argument'
        auto index = arrayPtr->getNextNode();
                     ~~~~~~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Array.cpp:191:27: error: no member named 'getNextNode' in 'llvm::Argument'
        auto newSize = arrayPtr->getNextNode();
                       ~~~~~~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Arith256.cpp:86:14: error: no member named 'getNextNode' in 'llvm::Argument'
        auto y = x->getNextNode();
                 ~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/BasicBlock.cpp:170:23: error: no member named 'getNextNode' in 'llvm::Argument'
        auto sizePtr = base->getNextNode();
                       ~~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Arith256.cpp:203:17: error: no member named 'getNextNode' in 'llvm::Argument'
        auto yArg = x->getNextNode();
                    ~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Arith256.cpp:301:14: error: no member named 'getNextNode' in 'llvm::Argument'
        auto y = x->getNextNode();
                 ~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Arith256.cpp:325:14: error: no member named 'getNextNode' in 'llvm::Argument'
        auto y = x->getNextNode();
                 ~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Arith256.cpp:369:14: error: no member named 'getNextNode' in 'llvm::Argument'
        auto y = x->getNextNode();
                 ~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Arith256.cpp:415:14: error: no member named 'getNextNode' in 'llvm::Argument'
        auto y = x->getNextNode();
                 ~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Arith256.cpp:441:14: error: no member named 'getNextNode' in 'llvm::Argument'
        auto y = x->getNextNode();
                 ~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Arith256.cpp:464:25: error: no member named 'getNextNode' in 'llvm::Argument'
                auto exponent = base->getNextNode();
                                ~~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Compiler.cpp:117:4: error: no matching constructor for initialization of 'IRBuilder' (aka 'IRBuilder<>')
                        IRBuilder{it}.CreateBr(nextBlock);
                        ^        ~~~~
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:540:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'llvm::ilist_iterator<llvm::BasicBlock>' to 'const llvm::IRBuilder<true, llvm::ConstantFolder,
      llvm::IRBuilderDefaultInserter<true> >' for 1st argument
class IRBuilder : public IRBuilderBase, public Inserter {
      ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:540:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'llvm::ilist_iterator<llvm::BasicBlock>' to
      'llvm::IRBuilder<true, llvm::ConstantFolder, llvm::IRBuilderDefaultInserter<true> >' for 1st argument
class IRBuilder : public IRBuilderBase, public Inserter {
      ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:550:12: note: candidate constructor not viable: no known conversion from 'llvm::ilist_iterator<llvm::BasicBlock>' to 'llvm::LLVMContext &' for 1st argument
  explicit IRBuilder(LLVMContext &C, MDNode *FPMathTag = nullptr,
           ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:560:12: note: candidate constructor not viable: no known conversion from 'llvm::ilist_iterator<llvm::BasicBlock>' to 'llvm::BasicBlock *' for 1st argument
  explicit IRBuilder(BasicBlock *TheBB, MDNode *FPMathTag = nullptr,
           ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:566:12: note: candidate constructor not viable: no known conversion from 'llvm::ilist_iterator<llvm::BasicBlock>' to 'llvm::Instruction *' for 1st argument
  explicit IRBuilder(Instruction *IP, MDNode *FPMathTag = nullptr,
           ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:554:12: note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
  explicit IRBuilder(BasicBlock *TheBB, const T &F, MDNode *FPMathTag = nullptr,
           ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:579:3: note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
  IRBuilder(BasicBlock *TheBB, BasicBlock::iterator IP,
  ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:544:3: note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
  IRBuilder(LLVMContext &C, const T &F, Inserter I = Inserter(),
  ^
/usr/local/opt/llvm/include/llvm/IR/IRBuilder.h:572:3: note: candidate constructor not viable: requires at least 3 arguments, but 1 was provided
  IRBuilder(BasicBlock *TheBB, BasicBlock::iterator IP, const T &F,
  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Compiler.cpp:128:42: error: no viable conversion from 'llvm::ilist_iterator<llvm::BasicBlock>' to 'llvm::BasicBlock *'
                                jumpTableInput->addIncoming(destIdx, it); // Fill up PHI node
                                                                     ^~
/usr/local/opt/llvm/include/llvm/IR/Instructions.h:2569:42: note: passing argument to parameter 'BB' here
  void addIncoming(Value *V, BasicBlock *BB) {
                                         ^
1 error generated.
make[2]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/GasMeter.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 24%] Building CXX object libethcore/CMakeFiles/ethcore.dir/ChainOperationParams.cpp.o
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Memory.cpp:36:25: error: no member named 'getNextNode' in 'llvm::Argument'
                auto blkOffset = mem->getNextNode();
                                 ~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Memory.cpp:110:20: error: no member named 'getNextNode' in 'llvm::Argument'
        auto index = mem->getNextNode();
                     ~~~  ^
/Users/aeufemio/projects/ethereum/webthree-umbrella/solidity/libethereum/evmjit/libevmjit/Optimizer.cpp:60:35: error: no viable conversion from 'llvm::ilist_iterator<llvm::BasicBlock>' to 'llvm::BasicBlock *'
                        retPhi->addIncoming(abortCode, bbIt);
                                                       ^~~~
/usr/local/opt/llvm/include/llvm/IR/Instructions.h:2569:42: note: passing argument to parameter 'BB' here
  void addIncoming(Value *V, BasicBlock *BB) {
                                         ^
1 error generated.
make[2]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/BasicBlock.cpp.o] Error 1
[ 24%] Building CXX object libethcore/CMakeFiles/ethcore.dir/Common.cpp.o
8 errors generated.
make[2]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/Arith256.cpp.o] Error 1
[ 25%] Building CXX object libethcore/CMakeFiles/ethcore.dir/CommonJS.cpp.o
[ 26%] Building CXX object libethcore/CMakeFiles/ethcore.dir/ICAP.cpp.o
1 error generated.
make[2]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/Optimizer.cpp.o] Error 1
[ 27%] Building CXX object libethcore/CMakeFiles/ethcore.dir/KeyManager.cpp.o
[ 28%] Building CXX object libethcore/CMakeFiles/ethcore.dir/Precompiled.cpp.o
1 error generated.
make[2]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/Ext.cpp.o] Error 1
[ 29%] Building CXX object libethcore/CMakeFiles/ethcore.dir/SealEngine.cpp.o
5 errors generated.
make[2]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/Array.cpp.o] Error 1
[ 29%] Building CXX object libethcore/CMakeFiles/ethcore.dir/Transaction.cpp.o
2 errors generated.
make[2]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/Compiler.cpp.o] Error 1
2 errors generated.
make[2]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/Memory.cpp.o] Error 1
make[1]: *** [evmjit/libevmjit/CMakeFiles/evmjit.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 30%] Linking CXX shared library libevmcore.dylib
[ 30%] Built target evmcore
[ 31%] Linking CXX shared library libethcore.dylib
[ 31%] Built target ethcore
make: *** [all] Error 2
```

--- a/evmjit/libevmjit/Arith256.cpp
+++ b/evmjit/libevmjit/Arith256.cpp
@@ -81,9 +81,10 @@ llvm::Function* generateLongMulFunc(char const* _funcName, llvm::IntegerType* _t
 	func->setDoesNotAccessMemory();
 	func->setDoesNotThrow();
 
-	auto x = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* x = &(*iter++);
 	x->setName("x");
-	auto y = x->getNextNode();
+	llvm::Argument* y = &(*iter);
 	y->setName("y");
 
 	auto entryBB = llvm::BasicBlock::Create(func->getContext(), "Entry", func);
@@ -198,10 +199,11 @@ llvm::Function* createUDivRemFunc(llvm::Type* _type, llvm::Module& _module, char
 	auto zero = llvm::ConstantInt::get(_type, 0);
 	auto one = llvm::ConstantInt::get(_type, 1);
 
-	auto x = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* x = &(*iter++);
 	x->setName("x");
-	auto yArg = x->getNextNode();
-	yArg->setName("y");
+	llvm::Argument* y = &(*iter);
+	y->setName("y");
 
 	auto entryBB = llvm::BasicBlock::Create(_module.getContext(), "Entry", func);
 	auto mainBB = llvm::BasicBlock::Create(_module.getContext(), "Main", func);
@@ -210,17 +212,17 @@ llvm::Function* createUDivRemFunc(llvm::Type* _type, llvm::Module& _module, char
 	auto returnBB = llvm::BasicBlock::Create(_module.getContext(), "Return", func);
 
 	auto builder = IRBuilder{entryBB};
-	auto yLEx = builder.CreateICmpULE(yArg, x);
+	auto yLEx = builder.CreateICmpULE(y, x);
 	auto r0 = x;
 	builder.CreateCondBr(yLEx, mainBB, returnBB);
 
 	builder.SetInsertPoint(mainBB);
 	auto ctlzIntr = llvm::Intrinsic::getDeclaration(&_module, llvm::Intrinsic::ctlz, _type);
 	// both y and r are non-zero
-	auto yLz = builder.CreateCall(ctlzIntr, {yArg, builder.getInt1(true)}, "y.lz");
+	auto yLz = builder.CreateCall(ctlzIntr, {y, builder.getInt1(true)}, "y.lz");
 	auto rLz = builder.CreateCall(ctlzIntr, {r0, builder.getInt1(true)}, "r.lz");
 	auto i0 = builder.CreateNUWSub(yLz, rLz, "i0");
-	auto y0 = builder.CreateShl(yArg, i0);
+	auto y0 = builder.CreateShl(y, i0);
 	builder.CreateBr(loopBB);
 
 	builder.SetInsertPoint(loopBB);
@@ -296,9 +298,10 @@ llvm::Function* Arith256::getUDiv256Func(llvm::Module& _module)
 	func->setDoesNotThrow();
 	func->setDoesNotAccessMemory();
 
-	auto x = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* x = &(*iter++);
 	x->setName("x");
-	auto y = x->getNextNode();
+	llvm::Argument* y = &(*iter);
 	y->setName("y");
 
 	auto bb = llvm::BasicBlock::Create(_module.getContext(), {}, func);
@@ -320,9 +323,10 @@ llvm::Function* createURemFunc(llvm::Type* _type, llvm::Module& _module, char co
 	func->setDoesNotThrow();
 	func->setDoesNotAccessMemory();
 
-	auto x = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* x = &(*iter++);
 	x->setName("x");
-	auto y = x->getNextNode();
+	llvm::Argument* y = &(*iter);
 	y->setName("y");
 
 	auto bb = llvm::BasicBlock::Create(_module.getContext(), {}, func);
@@ -364,9 +368,10 @@ llvm::Function* Arith256::getSDivRem256Func(llvm::Module& _module)
 	func->setDoesNotThrow();
 	func->setDoesNotAccessMemory();
 
-	auto x = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* x = &(*iter++);
 	x->setName("x");
-	auto y = x->getNextNode();
+	llvm::Argument* y = &(*iter);
 	y->setName("y");
 
 	auto bb = llvm::BasicBlock::Create(_module.getContext(), "", func);
@@ -410,9 +415,10 @@ llvm::Function* Arith256::getSDiv256Func(llvm::Module& _module)
 	func->setDoesNotThrow();
 	func->setDoesNotAccessMemory();
 
-	auto x = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* x = &(*iter++);
 	x->setName("x");
-	auto y = x->getNextNode();
+	llvm::Argument* y = &(*iter);
 	y->setName("y");
 
 	auto bb = llvm::BasicBlock::Create(_module.getContext(), {}, func);
@@ -436,9 +442,10 @@ llvm::Function* Arith256::getSRem256Func(llvm::Module& _module)
 	func->setDoesNotThrow();
 	func->setDoesNotAccessMemory();
 
-	auto x = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* x = &(*iter++);
 	x->setName("x");
-	auto y = x->getNextNode();
+	llvm::Argument* y = &(*iter);
 	y->setName("y");
 
 	auto bb = llvm::BasicBlock::Create(_module.getContext(), {}, func);
@@ -459,9 +466,10 @@ llvm::Function* Arith256::getExpFunc()
 		m_exp->setDoesNotThrow();
 		m_exp->setDoesNotAccessMemory();
 
-		auto base = &m_exp->getArgumentList().front();
+		auto iter = m_exp->arg_begin();
+		llvm::Argument* base = &(*iter++);
 		base->setName("base");
-		auto exponent = base->getNextNode();
+		llvm::Argument* exponent = &(*iter);
 		exponent->setName("exponent");
 
 		InsertPointGuard guard{m_builder};

--- a/evmjit/libevmjit/Array.cpp
+++ b/evmjit/libevmjit/Array.cpp
@@ -32,9 +32,10 @@ llvm::Function* Array::createArrayPushFunc()
 	func->setDoesNotThrow();
 	func->setDoesNotCapture(1);
 
-	auto arrayPtr = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* arrayPtr = &(*iter++);
 	arrayPtr->setName("arrayPtr");
-	auto value = arrayPtr->getNextNode();
+	llvm::Argument* value = &(*iter);
 	value->setName("value");
 
 	InsertPointGuard guard{m_builder};
@@ -82,11 +83,12 @@ llvm::Function* Array::createArraySetFunc()
 	func->setDoesNotThrow();
 	func->setDoesNotCapture(1);
 
-	auto arrayPtr = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* arrayPtr = &(*iter++);
 	arrayPtr->setName("arrayPtr");
-	auto index = arrayPtr->getNextNode();
+	llvm::Argument* index = &(*iter++);
 	index->setName("index");
-	auto value = index->getNextNode();
+	llvm::Argument* value = &(*iter);
 	value->setName("value");
 
 	InsertPointGuard guard{m_builder};
@@ -106,9 +108,10 @@ llvm::Function* Array::createArrayGetFunc()
 	func->setDoesNotThrow();
 	func->setDoesNotCapture(1);
 
-	auto arrayPtr = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* arrayPtr = &(*iter++);
 	arrayPtr->setName("arrayPtr");
-	auto index = arrayPtr->getNextNode();
+	llvm::Argument* index = &(*iter++);
 	index->setName("index");
 
 	InsertPointGuard guard{m_builder};
@@ -128,9 +131,10 @@ llvm::Function* Array::createGetPtrFunc()
 	func->setDoesNotThrow();
 	func->setDoesNotCapture(1);
 
-	auto arrayPtr = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* arrayPtr = &(*iter++);
 	arrayPtr->setName("arrayPtr");
-	auto index = arrayPtr->getNextNode();
+	llvm::Argument* index = &(*iter++);
 	index->setName("index");
 
 	InsertPointGuard guard{m_builder};
@@ -186,9 +190,10 @@ llvm::Function* Array::createExtendFunc()
 	func->setDoesNotThrow();
 	func->setDoesNotCapture(1);
 
-	auto arrayPtr = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* arrayPtr = &(*iter++);
 	arrayPtr->setName("arrayPtr");
-	auto newSize = arrayPtr->getNextNode();
+	llvm::Argument* newSize = &(*iter++);
 	newSize->setName("newSize");
 
 	InsertPointGuard guard{m_builder};

--- a/evmjit/libevmjit/BasicBlock.cpp
+++ b/evmjit/libevmjit/BasicBlock.cpp
@@ -165,17 +165,18 @@ llvm::Function* LocalStack::getStackPrepareFunc()
 	auto updateBB = llvm::BasicBlock::Create(func->getContext(), "Update", func);
 	auto outOfStackBB = llvm::BasicBlock::Create(func->getContext(), "OutOfStack", func);
 
-	auto base = &func->getArgumentList().front();
+	auto iter = func->arg_begin();
+	llvm::Argument* base = &(*iter++);
 	base->setName("base");
-	auto sizePtr = base->getNextNode();
+	llvm::Argument* sizePtr = &(*iter++);
 	sizePtr->setName("size.ptr");
-	auto min = sizePtr->getNextNode();
+	llvm::Argument* min = &(*iter++);
 	min->setName("min");
-	auto max = min->getNextNode();
+	llvm::Argument* max = &(*iter++);
 	max->setName("max");
-	auto diff = max->getNextNode();
+	llvm::Argument* diff = &(*iter++);
 	diff->setName("diff");
-	auto jmpBuf = diff->getNextNode();
+	llvm::Argument* jmpBuf = &(*iter);
 	jmpBuf->setName("jmpBuf");
 
 	InsertPointGuard guard{m_builder};

--- a/evmjit/libevmjit/Compiler.cpp
+++ b/evmjit/libevmjit/Compiler.cpp
@@ -109,12 +109,16 @@ void Compiler::resolveJumps()
 	// Iterate through all EVM instructions blocks (skip first one and last 4 - special blocks).
 	for (auto it = std::next(m_mainFunc->begin()), end = std::prev(m_mainFunc->end(), 4); it != end; ++it)
 	{
-		auto nextBlock = it->getNextNode(); // If the last code block, that will be "stop" block.
+		auto nextBlockIter = it;
+		++nextBlockIter; // If the last code block, that will be "stop" block.
+		auto currentBlockPtr = &(*it);
+		auto nextBlockPtr = &(*nextBlockIter);
+		
 		auto term = it->getTerminator();
 		llvm::BranchInst* jump = nullptr;
 
 		if (!term) // Block may have no terminator if the next instruction is a jump destination.
-			IRBuilder{it}.CreateBr(nextBlock);
+			IRBuilder{currentBlockPtr}.CreateBr(nextBlockPtr);
 		else if ((jump = llvm::dyn_cast<llvm::BranchInst>(term)) && jump->getSuccessor(0) == m_jumpTableBB)
 		{
 			auto destIdx = llvm::cast<llvm::ValueAsMetadata>(jump->getMetadata(c_destIdxLabel)->getOperand(0))->getValue();
@@ -125,10 +129,10 @@ void Compiler::resolveJumps()
 				jump->setSuccessor(0, bb);
 			}
 			else
-				jumpTableInput->addIncoming(destIdx, it); // Fill up PHI node
+				jumpTableInput->addIncoming(destIdx, currentBlockPtr); // Fill up PHI node
 
 			if (jump->isConditional())
-				jump->setSuccessor(1, nextBlock); // Set next block for conditional jumps
+				jump->setSuccessor(1, &(*nextBlockIter)); // Set next block for conditional jumps
 		}
 	}
 

--- a/evmjit/libevmjit/Ext.cpp
+++ b/evmjit/libevmjit/Ext.cpp
@@ -63,8 +63,9 @@ llvm::Value* Ext::getArgAlloca()
 	{
 		InsertPointGuard g{m_builder};
 		auto allocaIt = getMainFunction()->front().begin();
+		auto allocaPtr = &(*allocaIt);
 		std::advance(allocaIt, m_argCounter); // Skip already created allocas
-		m_builder.SetInsertPoint(allocaIt);
+		m_builder.SetInsertPoint(allocaPtr);
 		a = m_builder.CreateAlloca(Type::Word, nullptr, {"a.", std::to_string(m_argCounter)});
 	}
 	++m_argCounter;

--- a/evmjit/libevmjit/GasMeter.cpp
+++ b/evmjit/libevmjit/GasMeter.cpp
@@ -164,11 +164,12 @@ GasMeter::GasMeter(IRBuilder& _builder, RuntimeManager& _runtimeManager) :
 	auto updateBB = llvm::BasicBlock::Create(_builder.getContext(), "Update", m_gasCheckFunc);
 	auto outOfGasBB = llvm::BasicBlock::Create(_builder.getContext(), "OutOfGas", m_gasCheckFunc);
 
-	auto gasPtr = &m_gasCheckFunc->getArgumentList().front();
+	auto iter = m_gasCheckFunc->arg_begin();
+	llvm::Argument* gasPtr = &(*iter++);
 	gasPtr->setName("gasPtr");
-	auto cost = gasPtr->getNextNode();
+	llvm::Argument* cost = &(*iter++);
 	cost->setName("cost");
-	auto jmpBuf = cost->getNextNode();
+	llvm::Argument* jmpBuf = &(*iter);
 	jmpBuf->setName("jmpBuf");
 
 	InsertPointGuard guard(m_builder);

--- a/evmjit/libevmjit/Memory.cpp
+++ b/evmjit/libevmjit/Memory.cpp
@@ -31,15 +31,16 @@ llvm::Function* Memory::getRequireFunc()
 		func = llvm::Function::Create(llvm::FunctionType::get(Type::Void, argTypes, false), llvm::Function::PrivateLinkage, "mem.require", getModule());
 		func->setDoesNotThrow();
 
-		auto mem = &func->getArgumentList().front();
+		auto iter = func->arg_begin();
+		llvm::Argument* mem = &(*iter++);
 		mem->setName("mem");
-		auto blkOffset = mem->getNextNode();
+		llvm::Argument* blkOffset = &(*iter++);
 		blkOffset->setName("blkOffset");
-		auto blkSize = blkOffset->getNextNode();
+		llvm::Argument* blkSize = &(*iter++);
 		blkSize->setName("blkSize");
-		auto jmpBuf = blkSize->getNextNode();
+		llvm::Argument* jmpBuf = &(*iter++);
 		jmpBuf->setName("jmpBuf");
-		auto gas = jmpBuf->getNextNode();
+		llvm::Argument* gas = &(*iter);
 		gas->setName("gas");
 
 		auto preBB = llvm::BasicBlock::Create(func->getContext(), "Pre", func);
@@ -105,14 +106,16 @@ llvm::Function* Memory::createFunc(bool _isStore, llvm::Type* _valueType)
 	InsertPointGuard guard(m_builder); // Restores insert point at function exit
 
 	m_builder.SetInsertPoint(llvm::BasicBlock::Create(func->getContext(), {}, func));
-	auto mem = &func->getArgumentList().front();
+	
+	auto iter = func->arg_begin();
+	llvm::Argument* mem = &(*iter++);
 	mem->setName("mem");
-	auto index = mem->getNextNode();
+	llvm::Argument* index = &(*iter++);
 	index->setName("index");
 
 	if (_isStore)
 	{
-		auto valueArg = index->getNextNode();
+		llvm::Argument* valueArg = &(*iter);
 		valueArg->setName("value");
 		auto value = isWord ? Endianness::toBE(m_builder, valueArg) : valueArg;
 		auto memPtr = m_memory.getPtr(mem, m_builder.CreateTrunc(index, Type::Size));

--- a/evmjit/libevmjit/Optimizer.cpp
+++ b/evmjit/libevmjit/Optimizer.cpp
@@ -57,7 +57,8 @@ bool LongJmpEliminationPass::runOnFunction(llvm::Function& _func)
 		{
 			auto longjmp = term->getPrevNode();
 			assert(llvm::isa<llvm::CallInst>(longjmp));
-			retPhi->addIncoming(abortCode, bbIt);
+			auto bbPtr = &(*bbIt);
+			retPhi->addIncoming(abortCode, bbPtr);
 			llvm::ReplaceInstWithInst(term, llvm::BranchInst::Create(&exitBB));
 			longjmp->eraseFromParent();
 			modified = true;

